sfge/src/test/java/com/salesforce/graph/ops/expander/StackDepthLimitTest.java

```diff
@@ -0,0 +1,201 @@
+package com.salesforce.graph.ops.expander;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+import com.salesforce.TestRunner;
+import com.salesforce.TestUtil;
+import com.salesforce.config.SfgeConfigTestProvider;
+import com.salesforce.config.TestSfgeConfig;
+import com.salesforce.graph.symbols.apex.ApexStringValue;
+import com.salesforce.graph.visitor.PathVertexVisitor;
+import com.salesforce.graph.visitor.SystemDebugAccumulator;
+import java.util.List;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+public class StackDepthLimitTest {
+    private GraphTraversalSource g;
+    private static final int STACK_DEPTH_LIMIT_OVERRIDE = 10;
+
+    @BeforeEach
+    public void setup() {
+        this.g = TestUtil.getGraph();
+        SfgeConfigTestProvider.set(
+                new TestSfgeConfig() {
+                    @Override
+                    public int getStackDepthLimit() {
+                        return STACK_DEPTH_LIMIT_OVERRIDE;
+                    }
+                });
+    }
+
+    @AfterEach
+    public void cleanUp() {
+        SfgeConfigTestProvider.remove();
+    }
+
+    private static final String METHOD_NAME_BASE = "invokeLevel";
+    public static final String METHOD_INVOKE_LEVEL_TEMPLATE =
+            "  void "
+                    + METHOD_NAME_BASE
+                    + "%d() {\n"
+                    + "       "
+                    + METHOD_NAME_BASE
+                    + "%d();\n"
+                    + "   }\n";
+
+    @CsvSource({
+        "ExceedsLimit, " + (STACK_DEPTH_LIMIT_OVERRIDE + 2) + ", false",
+        "WithinLimit, " + (STACK_DEPTH_LIMIT_OVERRIDE - 2) + ", true"
+    })
+    @ParameterizedTest(name = "{displayName}: {0}")
+    public void testStackDepth(
+            String testName, int stackDepthToCreate, boolean expectToReachSystemDebug) {
+        // Since we already create two methods other than the dynamic ones, subtract them from
+        // required count
+        final int depthCountToAdd = stackDepthToCreate - 2;
+        final String sourceCode =
+                "public class MyClass {\n"
```

Comment:
You can wrap your code in `spotless:off` and `spotless:on` to make it immune to reformatting. I find it makes the source code in tests way easier to read.